presto-tpch/src/main/java/com/facebook/presto/tpch/TpchRecordSet.java

```diff
@@ -117,14 +141,16 @@ public Type getType(int field)
         @Override
         public boolean advanceNextPosition()
         {
-            if (closed || !rows.hasNext()) {
-                closed = true;
-                row = null;
-                return false;
+            while (!closed && rows.hasNext()) {
+                row = rows.next();
+                if (rowMatchesPredicate()) {
```

Comment:
> The intention of adding this was to have possibility to test how predicate pushdown to connector plays with cost calculator.

This is exactly the reason.

> Even though in case of TPC-H connector this pushdown is strictly artificial.

I think filtering is required in order to comply with connector layout interface. If connector specifies that it will enforce some constraint, then it should do the filtering.